src/snowflake/sqlalchemy/provision.py

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from sqlalchemy.testing.provision import set_default_schema_on_connection
+
+
+@set_default_schema_on_connection.for_db("snowflake")
+def _snowflake_set_default_schema_on_connection(cfg, dbapi_connection, schema_name):
```

Comment:
Could you explain what this is for? The connector should already be choosing the default schema if one isn't provided.

Reply:
this is only for test purpose required by the sql alchemy test suites. I've added a comment in the code